### PR TITLE
added Bogo sort and ternary search

### PR DESCRIPTION
**Pull request by aditya-nv-06**
Made the indentation changes

### CODE REVIEW

File: algorithm/sorting-algo-python/bogo_sort.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import random
```
Comment (FerdausPolok) [inline algorithm/sorting-algo-python/bogo_sort.py]: We dont have random here on this dir

File: algorithm/sorting-algo-python/bogo_sort.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        random.shuffle(arr)
+        attempts += 1
+    print(f"Array sorted in {attempts} attempts")
+    return arr
+
+
+    arr = [3, 2, 5, 1, 4]
+    print("Original array:", arr)
+    sorted_arr = bogo_sort(arr)
+    print("Sorted array:", sorted_arr)
```
Comment (FerdausPolok) [inline algorithm/sorting-algo-python/bogo_sort.py]: Please format with Prettier

File: algorithm/sorting-algo-python/bogo_sort.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    #Sort the array using Bogo Sort algorithm.
+    attempts = 0
+    while not is_sorted(arr):
+        random.shuffle(arr)
+        attempts += 1
+    print(f"Array sorted in {attempts} attempts")
+    return arr
+
+
+    arr = [3, 2, 5, 1, 4]
```
Comment (FerdausPolok) [inline algorithm/sorting-algo-python/bogo_sort.py]: Fix indentation please.

**Review by FerdausPolok — CHANGES_REQUESTED** [review]

**Comment by aditya-nv-06** [comment]
Made the changes and wrote the sort with the help of permutations .

File: algorithm/sorting-algo-python/bogo_sort.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+from itertools import permutations
```
Comment (FerdausPolok) [inline algorithm/sorting-algo-python/bogo_sort.py]: Same for this one. We don'tt have any libraries here in this repo. Maybe you should try doing everything raw/ manually here.

File: algorithm/searching-algo-python/ternary_search.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    return -1
+
+# Example usage
+arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+key = 5
+result = ternary_search(arr, 0, len(arr) - 1, key)
+if result != -1:
+        print(f"Element found at index {result}")
+else:
+        print("Element not found")
```
Comment (FerdausPolok) [inline algorithm/searching-algo-python/ternary_search.py]: EOL missing, please format your dicument with something like Prettier.

File: algorithm/searching-algo-python/ternary_search.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        elif key > arr[mid2]:
+            return ternary_search(arr, mid2 + 1, right, key)
+        else:
+            return ternary_search(arr, mid1 + 1, mid2 - 1, key)
+    
+    return -1
+
+# Example usage
+arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+key = 5
```
Comment (FerdausPolok) [inline algorithm/searching-algo-python/ternary_search.py]: Let's take these values from users.

**Review by FerdausPolok — CHANGES_REQUESTED** [review]

File: algorithm/searching-algo-python/ternary_search.py (hunk 6 above)
Comment (aditya-nv-06) [reply algorithm/searching-algo-python/ternary_search.py]: Ok

**Comment by aditya-nv-06** [comment]
Completed the changes

**Comment by aditya-nv-06** [comment]
Made all the changes.

File: algorithm/searching-algo-python/ternary_search.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+            return ternary_search(arr, mid1 + 1, mid2 - 1, key)
+    
+    return -1
+
+# Example usage
+arr = list(map(int, input("Enter the sorted array elements separated by spaces: ").split()))
+key = int(input("Enter the key to search for: "))
+result = ternary_search(arr, 0, len(arr) - 1, key)
+if result != -1:
+        print(f"Element found at index {result}")
```
Comment (FerdausPolok) [inline algorithm/searching-algo-python/ternary_search.py]: Could you please format your code with https://pypi.org/project/black/?
It seems like the indentation are not consistant. Thanks.

Comment (aditya-nv-06) [reply algorithm/searching-algo-python/ternary_search.py]: Ok

Comment (aditya-nv-06) [reply algorithm/searching-algo-python/ternary_search.py]: Corrected  the code

**Comment by aditya-nv-06** [comment]
Finally refactored the code

**Review by FerdausPolok — APPROVED** [review]

**Pull request merged**